src/txmempool.cpp

```diff
@@ -663,15 +663,19 @@ void CTxMemPool::removeForReorg(CChainState& active_chainstate, int flags)
                 }
             }
         }
-        if (!validLP) {
-            mapTx.modify(it, update_lock_points(lp));
-        }
     }
     setEntries setAllRemoves;
     for (txiter it : txToRemove) {
         CalculateDescendants(it, setAllRemoves);
     }
     RemoveStaged(setAllRemoves, false, MemPoolRemovalReason::REORG);
+    auto chain = active_chainstate.m_chain;
+    for (indexed_transaction_set::const_iterator it = mapTx.begin(); it != mapTx.end(); it++) {
+        LockPoints lp = it->GetLockPoints();
+        if (!TestLockPointValidity(chain, &lp)) {
```

Comment:
unrelated: Seems a bit confusing to pass a pointer here. Usually this means the memory it points to will be modified. However, in this case `lp` can be const.

Reply:
Actually bedf246f1e2497a3641093c6e8fa11fb34dddac4 is wrong.

`CheckSequenceLocks` modifies `lp`, so dropping it and using a fresh `lp` here from `GetLockPoints` is illegal.

Reply:
Oof, that's not good. Can be fixed by moving `update_lock_points` back to txmempool.h and updating lockpoints within the lambda (original approach in b5de6cc10bfbc8174936c3e8ae9418dedfb60000). Should I open a pull with that?

Reply:
> Actually [bedf246](https://github.com/bitcoin/bitcoin/commit/bedf246f1e2497a3641093c6e8fa11fb34dddac4) is wrong.
> 
> `CheckSequenceLocks` modifies `lp`, so dropping it and using a fresh `lp` here from `GetLockPoints` is illegal.

#23897 aims to make it less fragile in the future.